staging/src/k8s.io/csi-translation-lib/plugins/gce_pd.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	// GCEPDDriverName is the name of the CSI driver for GCE PD
+	GCEPDDriverName = "pd.csi.storage.gke.io"
+	// GCEPDInTreePluginName is the name of the intree plugin for GCE PD
+	GCEPDInTreePluginName = "kubernetes.io/gce-pd"
+
+	// Volume ID Expected Format
+	// "projects/{projectName}/zones/{zoneName}/disks/{diskName}"
+	volIDZonalFmt = "projects/%s/zones/%s/disks/%s"
+	// "projects/{projectName}/regions/{regionName}/disks/{diskName}"
+	volIDRegionalFmt   = "projects/%s/regions/%s/disks/%s"
+	volIDDiskNameValue = 5
+	volIDTotalElements = 6
+
+	// LabelZoneFailureDomain is the label on PVs indicating the zone they are provisioned in
+	LabelZoneFailureDomain = "failure-domain.beta.kubernetes.io/zone"
+	// LabelMultiZoneDelimiter separates zones for RePD volumes
+	LabelMultiZoneDelimiter = "__"
+	// UnspecifiedValue is used for an unknown zone string
+	UnspecifiedValue = "UNSPECIFIED"
+)
+
+// GCEPD handles translation of PV spec from In-tree GCE PD to CSI GCE PD and vice versa
+type GCEPD struct{}
```

Comment:
Make `AWSEBS`, `GCEPD`, etc. structs private (instead of public) and add a constructor methods that return the `InTreePlugin interface` so that consumers depend only on the interface:

```GO
func NewGCEPD(...) InTreePlugin {
  return &gcePD{...}
}

type gcePD struct{}
```